app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt

```diff
@@ -121,7 +167,38 @@ class StateFragmentPresenter @Inject constructor(
 
   private fun subscribeToCurrentState() {
     ephemeralStateLiveData.observe(fragment, Observer<EphemeralState> { result ->
-      logger.d("StateFragment", "getCurrentState: ${result.state.name}")
+      if (result.hasState()) {
+        itemList.clear()
+        currentEphemeralState.set(result)
+
+        updateDummyStateName()
+
+        val interactionId = result.state.interaction.id
+        val hasPreviousState = result.hasPreviousState
+        var hasNextState = false
+        hasGeneralContinueButton = false
+
+        if (!result.terminalState) {
+          if (result.stateTypeCase.number == EphemeralState.COMPLETED_STATE_FIELD_NUMBER
+            && !oldStateNameList.contains(currentEphemeralState.get()!!.state.name)
+          ) {
+            hasGeneralContinueButton = true
```

Comment:
I'm not sure I follow this logic--what exactly does ``hasGeneralContinueButton`` represent?

Reply:
@BenHenning This one is actually, quite tricky to explain in writing and it involves much more demo-based explanation. Still I will try to summarise this hear and if it does not clarify things we can discuss this in 1:1.

Assuming learner is on State-2 or 3 which has `TextInput` interaction-id. Now in this case learner submits an answer, so once answer is submitted correctly, we now display the `Continue` button to move to next. Note: We are not clicking on this `Continue` button for now.

Now in this same state, learner clicks on `Previous` button and again clicks on `Next` button to come to this state.

Now here is the tricky thing:
Main question is how do we decide whether we have to show the `Next` button or `Continue` button because the learner has already submitted the correct answer.

Here, this `hasGeneralContinueButton ` comes into picture, if this is true than we have to show `Continue` button to move to next-state and if this is `false` then we have to show `Next` button to move to next-state.

Does this help?

Reply:
If I understand correctly, this is only to solve the case where a correct answer has been submitted and we navigate away & then back to the most recent state?

If so, could we infer hasGeneralContinueButton based on whether the current state is COMPLETED and there's no next state? Perhaps what we really need here is a has_next_state similar to has_previous_state to solve this. This edge case where the latest state is also COMPLETED & then we navigate away didn't occur to me, so it seems like a potential issue in the domain layer.

If my suggested fix seems reasonable, then please file a new issue against me to fix that and add TODOs here to remove the UI-based workaround. I'll clean it up with a domain layer fix, but let's not block this entire PR on that specific detail.

Reply:
Added issue #257